code/Model/SiteTree.php

```diff
@@ -2959,17 +2977,28 @@ public function getControllerName()
             }
             // If we have a class of "{$ClassName}Controller" then we found our controller
             if (class_exists($candidate = sprintf('%sController', $class))) {
-                $controller = $candidate;
-                break;
+                return $candidate;
             } elseif (class_exists($candidate = sprintf('%s_Controller', $class))) {
                 // Support the legacy underscored filename, but raise a deprecation notice
                 Deprecation::notice(
                     '5.0',
                     'Underscored controller class names are deprecated. Use "MyController" instead of "My_Controller".',
                     Deprecation::SCOPE_GLOBAL
                 );
-                $controller = $candidate;
-                break;
+                return $candidate;
+            } elseif ($namespaceMap !== null) {
+                foreach ($namespaceMap as $pageNamespace => $controllerNamespace) {
```

Comment:
Add a check `$namespaceMap` is an associative array?

Reply:
Could do - it's likely to just fail on the `strpos` check though, which I think is acceptable. The config variable is called `namespace_mapping` after all, it does indicate an asssociative array should be set

Reply:
It wouldn't get to the `strpos` check if it wasn't an array, failing with `Invalid argument supplied for foreach()`

Reply:
Ah well if it's not an `array` at all, yeah - I thought you meant it if it was a sequential array.

So basically you are suggesting:

```diff
- } elseif ($namespaceMap !== null) {
+ } elseif (is_array($namespaceMap) {
```

is that right?

Reply:
Initially, I meant to check for an actual associative array as it's a map, but checking for an array as a minimum is a good-enough middle ground.